ecommerce-ui/src/listings.js

```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+
+class Listings extends React.Component{
+    render(){
+        const listings = this.props.listing.map((item, idx) => {
+            return (
+                    <div key={idx}>
```

Comment:
I would refactor this code into a reusable component ( for credit)

Reply:
Also don't forget to define propTypes

Reply:
Hi Este,

I managed to get shopping cart to work and have been working on testing.  My fail at .map which I assume is why you wanted me to refactor the code.  I am not exactly sure how to refactor the code considering the props are coming from app.js

Reply:
Hey Jeremy, 
So what I meant by refactor was that you should create a component for the listing instead of writing all the JSX inside the return. Part of the credit for this assignment was creating reusable components. Listings is more of a container component and is only used once. The individual cards for the listed properties are going to be reused multiple times on one page so it makes more sense to turn them into a reusable component. 
`return (<Listing key={idx} listingData={item} callback={this.props.someCallback} />)`
Something more along those lines. 

As far as testing goes the most important part is coding at least 1 snapshot test. The Rubric says one per component. But showing just one example for me is fine. Here is the in class code for reference : https://github.com/modern-web-application-uw19/react-testing/blob/master/src/App.test.js